File: examples/circles.cpp
#include <bimage.h>
#include "../twombly.hpp"

using namespace tw;

#define SIZE 800

int main(int argc, char *argv[]){
    bimage *im = bimageCreate(SIZE, SIZE, BIMAGE_U16 | 4);
    auto d = draw(SIZE, SIZE, (uint16_t*)im->data);

    d.antialias(false);

    for (size_t i = 0; i < 255; i += 20){
        d.ellipse(SIZE/2, SIZE/2, (SIZE/2)-10 - i, (SIZE/2)-10 - i);
        d.fill(Color(i, 0, 255 - i / 2));
        d.paint();
    }

    bimageSave(im, "circles.tiff");
    bimageDestroy(&im);
    return 0;
}

File: examples/circles.py
from twombly import *
from skimage.io import imsave

SIZE=800
im = new_image(SIZE, SIZE)
d = draw(im)
#d.set_antialias(False)

for i in xrange(0, 255, 20):
    d.new_path()
    d.ellipse(SIZE/2, SIZE/2, SIZE/2-10 - i, SIZE/2-10 -i)
    d.set_color(i, 0, 255 - i / 2)
    d.fill()

imsave("circles.png", im)

